packages/bcer-data-portal/app/src/hooks/useLeaflet.tsx

```diff
@@ -0,0 +1,338 @@
+import React, { useContext, useEffect, useState } from 'react';
+import L from 'leaflet';
+import GeoJSON from 'geojson';
+
+import {
+  BCDirectionData,
+  BCGeocoderAutocompleteData,
+  BusinessLocation,
+  LocationConfig,
+  RouteOptions,
+} from '@/constants/localInterfaces';
+import useLocation from './useLocation';
+import { AppGlobalContext } from '@/contexts/AppGlobal';
+import { BcRouteLinkBuilder } from '@/util/bcRouteLink.util';
+import { GoogleMapLinkBuilder } from '@/util/googleMapLink.util';
+
+function useLeaflet(locationIds: string, config: LocationConfig) {
+  const [appGlobal, setAppGlobalContext] = useContext(AppGlobalContext);
+
+  const [routeData, setRouteData] = useState<BCDirectionData>();
+  const [directionError, setDirectionError] = useState<any>();
+
+  const {
+    selectedLocations,
+    setSelectedLocations,
+    removeSelectedLocationHandler,
+    addLocationToSelectedHandler,
+  } = useLocation(locationIds);
+
+  const [map, setMap] = useState<L.Map>();
+  const [markers, setMarkers] = useState<L.Marker[]>([]);
+  const [geoJsonData, setGeoJsonData] = useState<L.GeoJSON>();
+  const [startingLocation, setStartingLocation] =
+    useState<BCGeocoderAutocompleteData>();
+
+  // Initial routing options in the map controller
+  const initialRoutingOptions: RouteOptions = {
+    option: 'fastest',
+    roundTrip: false,
+    optimizeOrder: false,
+    ferrySchedule: false,
+    timeDependent: false,
+    traffic: false,
+    turnRestriction: false,
+    events: false,
+    crossingCost: false,
+    globalDistortionField: false,
+    turnCost: false,
+    localDistortionField: false,
+  };
+
+  const [routeOptions, setRouteOptions] = useState<RouteOptions>({
+    ...initialRoutingOptions,
+  });
+
+  /**
+   * Once Map is rendered in the dom this method must be called
+   * to set up map properties
+   */
+  const onRender = () => {
+    let m = L.map('map').setView([51.4, -116.9], 6);
```

Comment:
Just curious, I assume this correspond to a view of BC?

Reply:
yeah. pretty much contains all of BC!